### PR TITLE
[4.0] Change com_templates to services

### PR DESCRIPTION
**Pull request by laoneo**
Moves com_templates to the service based extension setup.

See #20217 for details.

### CODE REVIEW

**Comment by brianteeman** [comment]
Cant test on the front end because of #20356

**Comment by laoneo** [comment]
But you can test on the back end? Front end is routing through com_config thats why #20356 will be fixed there too.

**Comment by brianteeman** [comment]
and i am saying that AFTER applying this PR i still get the error in #20356

**Comment by brianteeman** [comment]
ah i see what you meant now. as i said in #20620 it cant be tested with patchtester so it makes it harder to test - dont have time to do that right now

File: libraries/extensions.classmap.php [hunk 1]
```diff
@@ -14,3 +14,6 @@
 JLoader::registerAlias('FieldsListPlugin', '\\Joomla\\Component\\Fields\\Administrator\\Plugin\\FieldsListPlugin', '4.0');
 
 JLoader::registerAlias('TagsTableTag',     '\\Joomla\\Component\\Tags\\Administrator\\Table\\TagTable', '4.0');
+
+JLoader::registerAlias('TemplatesHelper',  '\\Joomla\\Component\\Templates\\Administrator\\Helper\\TemplatesHelper', '5.0');
```
Comment (wilsonge) [inline libraries/extensions.classmap.php]: I'm not a fan of this. These un-namespaced classes are so generically named - I think rather just have the empty deprecated classes that people need to require :/ @mbabker do you have any strong opinions?

**Comment by infograf768** [comment]
When viewing `administrator/index.php?option=com_templates&view=templates`
I get a `500 JHtml templates not found.`
@laoneo

**Comment by anuragteapot** [comment]
@infograf768 Please test #20649

**Pull request merged**